components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/RegistryHelper.java

```diff
@@ -69,4 +69,18 @@ public static String getSystemDependentPath(String path) {
         return path.replace(URL_SEPARATOR_CHAR, File.separatorChar);
     }
 
+    /**
+     * Check whether the directory is empty
+     *
+     * @param directory  directory path
+     */
+    public static boolean isDirectoryEmpty (String directory) {
+        File file = new File(directory);
+        if (file.isDirectory()) {
+            return file.list().length == 0;
+        } else {
+            return false;

```

Comment:
Should we return an error since we are expecting 'directory' to be a directory?

Reply:
The method is to check whether the directory is empty. We don't want to fail the flow, if we are checking empty when it is not a directory, for an example if it is a file. It just says false. WDYT?